src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -0,0 +1,242 @@
+
+  const handleError = (
+    fieldValue: string,
+    fieldVisited: boolean,
+  ) => {
+    return ((fieldValue.length === 0) && fieldVisited);
+  };
+
+  const resetError = (
+    setFieldValue: Dispatch<SetStateAction<string>>,
+    setFieldVisited: Dispatch<SetStateAction<boolean>>,
+  ) => {
+    setFieldValue('');
+    setFieldVisited(false);
+  };
+
+  const handleNewCommentId = (postId: number) => {
+    const postComments = comments.filter(note => note.postId === postId);
+    const maxCommentId = Math.max(...postComments.map(el => el.id));
+
+    setNewCommentId(maxCommentId + 1);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    handleNewCommentId(selectedPost.id);
+
+    if (!name.trim() || !email.trim() || !body.trim()) {
+      setNameVisited(true);
+      setEmailVisited(true);
+      setCommentVisited(true);
+
+      return;
+    }
+
+    onAdd({
+      id: newCommentId,
+      name,
+      email,
+      body,
+      postId: selectedPost.id,
+    });
+
+    setBody('');
+    setCommentVisited(false);
+  };
+
+  const clearSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    resetError(setName, setNameVisited);
+    resetError(setEmail, setEmailVisited);
+    resetError(setBody, setCommentVisited);
+  };
+
+  return (
+    <form
+      data-cy="NewCommentForm"
+      onSubmit={handleSubmit}
+      onReset={clearSubmit}
+    >
+      <div className="field" data-cy="NameField">
+        <label className="label" htmlFor="comment-author-name">
+          Author Name
+        </label>
+
+        <div className="control has-icons-left has-icons-right">
+          <input
+            type="text"
+            name="name"
+            id="comment-author-name"
+            placeholder="Name Surname"
+            className={classNames('input', {
+              'is-danger': handleError(name, nameVisited),
+            })}
+            value={name}
+            onChange={(e) => handleChange(e, setName, setNameVisited)}
+            onBlur={() => handleBlur(name, setNameVisited)}
+          />
+
+          <span className="icon is-small is-left">
+            <i className="fas fa-user" />
+          </span>
+
+          {handleError(name, nameVisited) && (
+            <>
+              <span
+                className="icon is-small is-right has-text-danger"
+                data-cy="ErrorIcon"
+              >
+                <i className="fas fa-exclamation-triangle" />
+              </span>
+            </>
+          )}
+        </div>
+
+        {handleError(name, nameVisited) && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Name is required
+          </p>
+        )}
+      </div>
+
+      <div className="field" data-cy="EmailField">
+        <label className="label" htmlFor="comment-author-email">
+          Author Email
+        </label>
+
+        <div className="control has-icons-left has-icons-right">
+          <input
+            type="text"
```

Comment:
The 'type' of the email input should be 'email' instead of 'text'. This will provide automatic validation for the email field.